rust/extractor/src/rust_analyzer.rs

```diff
@@ -188,10 +189,25 @@ fn from_utf8_lossy(v: &[u8]) -> (Cow<'_, str>, Option<SyntaxError>) {
     (Cow::Owned(res), Some(error))
 }
 
+fn canonicalize_if_on_windows(path: &Path) -> Option<PathBuf> {
+    if cfg!(windows) {
+        dunce::canonicalize(path).ok()
+    } else {
+        None
+    }
+}
+
 pub(crate) fn path_to_file_id(path: &Path, vfs: &Vfs) -> Option<FileId> {
-    Utf8PathBuf::from_path_buf(path.to_path_buf())
-        .ok()
-        .and_then(|x| AbsPathBuf::try_from(x).ok())
-        .map(VfsPath::from)
-        .and_then(|x| vfs.file_id(&x))
+    // There seems to be some flaky inconsistencies around UNC paths on Windows, so if we fail to
+    // find the file id for a UNC path like that, we try to canonicalize it using dunce then.
```

Comment:
```suggestion
    // There seems to be some flaky inconsistencies around UNC paths on Windows, so if we fail to
    // find the file id for a UNC path, we try with a canonicalized (by dunce) path as well.
```

Reply:
Makes sense! I'll do a follow up though on Monday, I prefer to not have the flakiness in for the night lies 🙂